src/Catty/ViewController/Upload/UploadViewController.swift

```diff
@@ -259,171 +203,32 @@ class UploadViewController: UIViewController {
-
-                    if statusCode == self.statusCodeOK {
-                        debugPrint("Upload successful")
-
-                        //Set unique Project-ID received from server
-                        if let aTag = dictionary?[self.projectIDTag], let project = self.project {
-                            let projectId = "\(aTag)"
-
-                            DispatchQueue.main.async(execute: {
-                                self.project?.rename(toProjectName: project.header.programName, andProjectId: projectId, andShowSaveNotification: true)
-                            })
-                        }
-
-                        //Set new token but when? everytime is wrong
-                        if let aTag = dictionary?[self.tokenParameterTag] {
-                            JNKeychain.saveValue(aTag, forKey: kUserLoginToken)
-                        }
-
-                        DispatchQueue.main.async(execute: {
-                            self.dismissView()
-                        })
-                    } else {
-                        let serverResponse = dictionary?[self.answerTag] as? String
-                        debugPrint("Error: ", serverResponse ?? "undefined")
-                        DispatchQueue.main.async(execute: {
-                            if serverResponse != nil {
-                                Util.alert(withText: serverResponse)
-                            } else {
-                                Util.alert(withText: kLocalizedUploadProblem)
-                            }
-                        })
-
-                        if statusCode == self.statusCodeTokenWrong {
-                            //Token not valid
-                            UserDefaults.standard.set(false, forKey: kUserIsLoggedIn)
-
-                            var viewArray: [AnyHashable]?
-                            if let aControllers = self.parent?.navigationController?.viewControllers {
-                                viewArray = aControllers
-                            }
-                            viewArray?.removeLast()
-                            var newViewArray: [AnyHashable]?
-                            if let anArray = viewArray {
-                                newViewArray = anArray
-                            }
-                            if let anArray = newViewArray as? [UIViewController] {
-                                self.parent?.navigationController?.setViewControllers(anArray, animated: true)
-                            }
-                        }
-                    }
-                }
-            }
-            dataTask.resume()
-        } else {
-            debugPrint("Could not build checksum")
-            enableUploadView()
-            Util.alert(withText: kLocalizedUploadProblem)
+        if let project = project, let uploader = self.uploader {
+            uploader.upload(project: project,
+                            completion: { error in
```

Comment:
Please add an optional `projectId` to the completion handler in the `StoreProjectUploader` which is passed along in case of success (retrieved from the server response).